influxdb3_write/src/persister.rs

```diff
@@ -120,25 +120,23 @@ impl Persister {
         &self.host_identifier_prefix
     }
 
-    /// Loads the most recently persisted catalog from object storage, and uses it to initialize
-    /// a [`LastCacheProvider`].
-    ///
-    /// This is intended to be used on server start.
-    pub async fn load_last_cache_and_catalog(
-        &self,
-        host_id: Arc<str>,
-    ) -> Result<(LastCacheProvider, Catalog)> {
-        match self.load_catalog().await? {
-            Some(c) => Ok((
-                LastCacheProvider::new_from_catalog(&c.catalog)?,
-                Catalog::from_inner(c.catalog),
-            )),
+    /// Try loading the catalog, if there is no catalog generate new
+    /// instance id and create a new catalog and persist it immediately
+    pub async fn load_or_create_catalog(&self) -> Result<Catalog> {
+        let catalog = match self.load_catalog().await? {
```

Comment:
@pauldix This is slightly different to what has been mentioned in the [issue](https://github.com/influxdata/influxdb/issues/25346), I'm reusing the `load_catalog` method which was already in use to determine whether to generate instance id or not. If this is not ideal, happy to rework this PR.